syntax = "proto2";
package final_project;

enum PhoneType 
{
    MOBILE = 0;
    HOME = 1;
    WORK = 2;
}

message TClientReq
{
	required int32 id = 1;
	required string name = 2;
}

message TClientRsp
{
	required int32 id = 1;
	required string file_name = 2;
	required string data = 3;
}

message TServerReq
{
	required int32 id = 1;
	required PhoneType type = 2; 
}

message TServerRsp
{
	required int32 id = 1;	
}

message TData 
{
	oneof Msg
	{
		TClientReq clientReq = 1;
		TClientRsp clientRsp = 2;
		TServerReq serverReq = 3;
		TServerRsp serverRsp = 4;
	}	
	
}
